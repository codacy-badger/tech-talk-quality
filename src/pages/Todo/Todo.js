import React, { useState } from 'react'

import App from '../../containers/App'

import styles from './Todo.module.css'

const Todo = () => {
  const [tasks, updateTasks] = useState([])
  const [task, updateTask] = useState('')

  const handleFormSubmit = event => {
    event.preventDefault()
    updateTasks(oldTasks => [...oldTasks, task])
    updateTask('')
  }

  const handleInputChange = event => updateTask(event.target.value)

  return (
    <App>
      <h1 className={styles.title} data-test="todo-title">
        Lista de tarefas
      </h1>
      <form className={styles.form} data-test="todo-form" onSubmit={handleFormSubmit}>
        <input
          className={styles.field}
          data-test="todo-field"
          onChange={handleInputChange}
          value={task}
        />
        <button className={styles.button} data-test="todo-button">
          Adicionar
        </button>
      </form>
      <table className={styles.table} data-test="todo-table">
        <thead>
          <tr>
            <th>Tarefa</th>
          </tr>
        </thead>
        <tbody>
          {tasks.length > 0 ? (
            tasks.map((t, i) => (
              <tr data-test={i} key={i}>
                <td data-test={t}>{t}</td>
              </tr>
            ))
          ) : (
            <tr>
              <td>Tabela vazia</td>
            </tr>
          )}
        </tbody>
      </table>
    </App>
  )
}

export default Todo
