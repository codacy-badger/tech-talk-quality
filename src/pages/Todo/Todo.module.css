.button {
  background-color: #0c9;
  border: none;
  border-radius: 2px;
  color: #f2f2f2;
  padding: 0.5rem 0.75rem;
  margin-left: 0.5rem;
}

.field {
  border: 1px solid #dcdcdc;
  border-radius: 4px;
  flex: 1;
  color: #595959;
  padding: 0.5rem;
}

.field:focus {
  border-color: #09f;
  outline: none;
}

.form {
  display: flex;
  margin: 1rem 0;
}

.table {
  border-collapse: collapse;
  width: 100%;
}

.table thead th {
  background-color: #fff;
  font-size: 1.1rem;
  text-align: left;
  padding: 0.5rem 0.25rem;
}

.table tbody tr:nth-child(even) {
  background-color: #e6e6e6;
}

.table tbody td {
  padding: 0.25rem;
}

.title {
  margin: 0;
}
