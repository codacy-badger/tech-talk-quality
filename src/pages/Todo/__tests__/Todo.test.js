import { render } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'

import Todo from '../Todo'

describe('Tests for Todo page', () => {
  it('Should render without crash', () => {
    const tree = renderer.create(<Todo />)
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('Should render title', () => {
    const { getByText } = render(<Todo />)
    expect(getByText('Lista de tarefas')).toBeInTheDocument()
  });
})
