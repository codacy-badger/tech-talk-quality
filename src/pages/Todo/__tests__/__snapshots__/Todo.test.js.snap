// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Todo page Should render withou crash 1`] = `
<main
  className="container"
>
  <div
    className="center"
  >
    <h1
      className="title"
      data-test="todo-title"
    >
      Lista de tarefas
    </h1>
    <form
      className="form"
      data-test="todo-form"
      onSubmit={[Function]}
    >
      <input
        className="field"
        data-test="todo-field"
        onChange={[Function]}
        value=""
      />
      <button
        className="button"
        data-test="todo-button"
      >
        Adicionar
      </button>
    </form>
    <table
      className="table"
      data-test="todo-table"
    >
      <thead>
        <tr>
          <th>
            Tarefa
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Tabela vazia
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</main>
`;
